### PR TITLE
Card# get appeal answer from gmail api

### PR DESCRIPTION
**Pull request by Sorey**

### CODE REVIEW

File: app/controllers/towns_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class TownsController < ApplicationController
   before_action :set_town, only: [:show, :edit, :update, :destroy, :export_town_in_xls]
+  before_action :access_user, only: [:index, :show, :edit, :update, :destroy]
```
Comment (Quicks) [inline app/controllers/towns_controller.rb]: maybe we should use except?

File: app/controllers/repairing/appeals_controller.rb [hunk 2]
```diff
@@ -1,23 +1,38 @@
 module Repairing
   class AppealsController < ApplicationController
     layout 'application_admin'
-    before_action :access_user, only: [:index, :show, :edit, :update, :destroy, :disapprove_form, :disapprove]
-    before_action :set_repairing_appeal, only: [:show, :edit, :update, :destroy, :approve, :disapprove_form, :disapprove]
+    before_action :access_user, only: [:edit, :update, :destroy, :disapprove_form, :disapprove]
```
Comment (Quicks) [inline app/controllers/repairing/appeals_controller.rb]: Maybe we should use except?

File: app/controllers/repairing/appeals_controller.rb [hunk 3]
```diff
@@ -112,6 +127,11 @@ def access_user
       redirect_to root_url, alert: t('export_budgets.notice_access')
     end
 
+    def access_user_by_town
+      return if current_user && current_user.has_any_role?(:admin, :city_authority, :central_authority)
```
Comment (Quicks) [inline app/controllers/repairing/appeals_controller.rb]: What the diif beetween access_user method?

**Comment by Sorey** [comment]
1) :only more correct for me
2) because some methods used only admin, other methods used admin,
city_authority and central authority and in access_approve_user every user
can auto-send appeal to town email


2018-06-15 21:02 GMT+03:00 Alexandro <notifications@github.com>:

**Pull request merged**